libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/SslFileUtil.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.ssl;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.file.AccessDeniedException;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.Path;
+import java.security.AccessControlException;
+import java.security.GeneralSecurityException;
+import java.security.UnrecoverableKeyException;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Utility methods for common file handling in SSL configuration
+ */
+final class SslFileUtil {
+
+    static String pathsToString(List<Path> paths) {
+        return paths.stream()
+            .map(Path::toAbsolutePath)
+            .map(Object::toString)
+            .collect(Collectors.joining(","));
+    }
+
+    static SslConfigException ioException(String fileType, List<Path> paths, IOException cause) {
+        return ioException(fileType, paths, cause, null);
+    }
+
+    static SslConfigException ioException(String fileType, List<Path> paths, IOException cause, String detail) {
```

Comment:
One of the big things we did to try and make X-Pack SSL setup easier was write clear & explicit error messages for all the common problems that people run into:
- File not found
- File permissions wrong
- File in wrong directory
- Missing password
- Wrong password

This class replicates those messages into ssl-config and centralises them.
Some of the messages already existed in this library, but the X-Pack ones are encountered more frequently, and have been better turned for readability.

That said, I did try and improve some of them along the way.